### ISSUE TITLE
Question: Image size parameter

### ISSUE THREAD

**Issue by zamponotiropita** [matterport/Mask_RCNN#435]
This is not an issue but a question.

How does the image size affect the accuracy of the results? I assume that with lower resolution we loose details and we can detect less features.
Does it make sense to continue training only the heads with pre-trained weights loaded from cocos with a different image resolution? I was thinking that since that model with pre-trained weights was trained with images of a certain resolution, there would be suboptimal results with features extracted from lower-resolution images.

**Comment by stanathong** [comment]
I have recently experimented this myself and could possibly answer both of your questions.
Based on my results, training with higher resolution images, the network tends to produce neat boundary results. 

What I have done is that I kept the default input size unchanged i.e. 1024x1024. My input images are of size 1920x1080. For the first experiment, I feed the images as they are in which the internal process will resize them to 1024x1024 with preserved aspect ratio. Here, my images are smaller resolution, resized by a ratio of 1024/1920. The second experiment, without modifying the network config parameters, I cropped the image to 1024x1024 and fed them to the network. Hence, this time the images are of higher resolution (than the first experiment). I found the second experiment produce a much neat segmentation boundary.

Also, I do find that training all layers give a better result than training head only for both of my experiments.

**Comment by waleedka** [comment]
@stanathong That makes sense. Thanks for sharing. 

Since you mentioned cropping the images, the new image resizing modes might be useful. For training, you can set `IMAGE_RESIZE_MODE = "crop"` and it will crop the images like you did. And, then, in the inference mode you set `IMAGE_RESIZE_MODE = "pad64"` and it runs the prediction on the full image 1920x1080 without resizing.

**Comment by stanathong** [comment]
@waleedka Glad to hear that the image resizing mode exists. I did modify the code to crop my images on the fly in which I could reassure that the objects are perfectly inside the crop image.
Thanks for letting me know and I will check it out.

Also, thanks very much for making this code public. It's awesome and very helpful.

**Comment by zamponotiropita** [comment]
@stanathong thanks for your inside!

**Comment by nemonameless** [comment]
@waleedka @stanathong @zamponotiropita    hello, I  set IMAGE_RESIZE_MODE = "pad64" to train my images data, I just want my image to be rectangle and  not square,  such as 1280*960, but I find the mrcnn_class_loss descends quickly to under 0.1 ,  but my learning rate is 0.001, which is small enough.  Even if I set IMAGE_RESIZE_MODE = "pad64", and let IMAGE_MIN_DIM = 960, IMAGE_MAX_DIM = 960,  mrcnn_class_loss  also  descends quickly to under 0.1 ,  is it overfiting? 

how does the input size affect the class_loss ? and it seems that input size does not affect the box_loss, why?

And if I changed  to  IMAGE_RESIZE_MODE = "square" ,to be 1280*1280, it performs well . but I still want my input image to be rectangle, such as 1280*960 , then what can I do ?
